Services/DeviceModels.cs

```diff
@@ -81,19 +98,70 @@ public DeviceModel Get(string id)
             throw new ResourceNotFoundException();
         }
 
-        private List<string> GetDeviceModelFiles()
+        /// <summary>
+        /// Create a device model.
+        /// </summary>
+        public async Task<DeviceModel> InsertAsync(DeviceModel deviceModel)
+        {
+            if (this.CheckDeviceModelExistence(deviceModel.Id))
+            {
+                throw new ConflictingResourceException("Device model with id '" + deviceModel.Id + "' already exists!");
+            }
+
+            try
+            {
+                var result = await this.customDeviceModels.InsertAsync(deviceModel);
+                deviceModel.ETag = result.ETag;
+            }
+            catch (Exception e)
+            {
+                this.log.Error("Unable to create a new device model", () => new { Exception = e });
```

Comment:
is it ok to swallow the exception and return a device model at line 121? what is the model going to be if the insert fails?

Reply:
should be something like:
```
throw new ExternalDependencyException("Failed to insert", e);

Reply:
Done.